src/iiif/queue_zip_consumer.py

```diff
@@ -0,0 +1,135 @@
+import json
+import logging
+import os
+import time
+
+import timeout_decorator
+from django.conf import settings
+from django.template.loader import render_to_string
+
+from iiif import authentication, image_server, mailing, utils, zip_tools
+from iiif.metadata import get_metadata
+from iiif.utils_azure import (
+    create_storage_account_temp_url,
+    get_queue_client,
+    store_object_on_storage_account,
+)
+from main import settings
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+class TimeoutError(Exception):
+    pass
+
+
+class AzureZipQueueConsumer:
+    # Be careful with the visibility timeout! If the message is still processing when the visibility timeout
+    # expires, the message will be put back on the queue and will be processed again. This can lead to duplicate
+    # messages!!! Always use a timeout decorator to prevent this.
+    # an hour, because some zips can simply be very very large
+    MESSAGE_VISIBILITY_TIMEOUT = 3600
+
+    def __init__(self, end_at_empty_queue=False):
+        self.queue_client = get_queue_client()
+        self.end_at_empty_queue = end_at_empty_queue
+
+    def get_queue_length(self):
+        properties = self.queue_client.get_queue_properties()
+        count = properties.approximate_message_count
+        return count
+    
+    def run(self):
+        while True:
+
+            count = self.get_queue_length()
+            message_iterator = None
+
+            if self.end_at_empty_queue:
+                message_iterator = [m for m in self.queue_client.receive_messages(messages_per_page=10, visibility_timeout=5)]
+                if count == 0 or len(message_iterator) == 0:
+                    break
```

Comment:
Sanity check:
- the comment above says to always set the visibility_timeout but you only provide 5 here

Reply:
This part is only for testing purposes. I added a comment in the code.